File: raw_data_processing/AZFP_data_by_day.py

# Required libraries
import os
import shutil

# Directory where the AZFP monthly data are stored
dep_name = input("Enter the full name of the deployment: ")
basedir = os.path.join("H:/dm1679/Data/Glider Data/",dep_name,"AZFP Data","")

for folder in next(os.walk(basedir))[1]:
    datadir = os.path.join(basedir,folder,"")

    for file in os.listdir(datadir): # For each file in the month's directory
        if file[-3:] == 'LOG': # Because the log files and AZFP files have slightly different naming conventions, need to check what kind of file it is first
            folder_name = file[6:8] # Gets the day of the month from how the log file is named, which will be our folder name
            # print(f'dir name: {folder_name}')

            folder_path = datadir + folder_name # Gets the path of the day folder
            # print(f'folder path = {folder_path}')

            if not os.path.exists(folder_path): # If the folder doesn't already exist...
                os.makedirs(folder_path) # Make a new folder for the day

            if os.path.exists(folder_path): # If the folder DOES exist...
                file_path = datadir + file # Get the full path to our file
                # print(f'file path: {file_path}')
                shutil.move(file_path, folder_path) # Move the file into the folder

        elif not os.path.isdir(datadir + file) and file[-3:] != 'cfg': # Steps below are similar to above but for the proper acoustic files
            folder_name = file[4:6]
            # print(f'dir name: {folder_name}')
            folder_path = datadir + folder_name
            # print(f'folder path = {folder_path}')

            if not os.path.exists(folder_path):
                os.makedirs(folder_path)

            if os.path.exists(folder_path):
                file_path = datadir + file
                shutil.move(file_path, folder_path)

        elif file[-3:] == 'cfg': # cfg is our configuration file and needs to be copied to every folder
            for fldr in os.listdir(datadir)[:-1]: # once we reach this step, we should have nothing but the day folders and the cfg file remaining; this gets the day folder names
                file_path = datadir + file
                folder_path = datadir + fldr
                # print(folder_path)
                shutil.copy2(file_path, folder_path) # copy the cfg file into every day folder
            os.remove(file_path) # and remove it from the month folder

File: raw_data_processing/AZFP_Data_By_Day.R
rm(list = ls())

library(tidyverse)
library(R.utils)
library(tcltk)

sourceDirectory(
  "H:/dm1679/Code/R_Functions",
  modifiedOnly = F
)

glider_dep = choose_directory(caption = "Select the glider deployment directory folder")
azfp_dep = paste0(glider_dep, "/AZFP Data/") # Raw AZFP data directory, in folders by month

monthly_folders = list.dirs(azfp_dep, recursive = F)

for(f in 1:length(monthly_folders)) { # for each month
  data_dir = monthly_folders[f]
  
  files = list.files(data_dir, recursive = T, include.dirs = F, full.names = F) # get all the files
  
  for(g in 1:length(files)) { # for each file
    current_file = files[g]
    
    if(substr(current_file, nchar(current_file)-2, nchar(current_file)) == "LOG") {
    # Each file type has slightly different naming conventions/lengths, so need
    # to check what sort of file it is before proceeding
      
    folder_name = substr(current_file, 7, 8) # get the day of the month
    
    create_dir(paste0(data_dir, "/", folder_name)) # create the daily folder, if it does not already exist
    
    file.rename(from = paste0(data_dir, "/", current_file),
                to = paste0(data_dir, "/", folder_name, "/", current_file))
    } else if(substr(current_file, nchar(current_file) - 2, nchar(current_file)) != "cfg") {
      
      folder_name = substr(current_file, 5, 6)
      
      create_dir(paste0(data_dir, "/", folder_name))
      
      file.rename(from = paste0(data_dir, "/", current_file),
                  to = paste0(data_dir, "/", folder_name, "/", current_file))
    } else if(substr(current_file, nchar(current_file)-2, nchar(current_file)) == "cfg") {
      # The cfg (config file) has to be copied into each daily folder
      
     daily_folders = list.dirs(data_dir, recursive = F)
     
     for(h in 1:length(daily_folders)) {
       file.copy(from = paste0(data_dir, "/", current_file),
                 to = paste0(daily_folders[h], "/", current_file))
     }
     file.remove(paste0(data_dir, "/", current_file))
    }
  }
}
